src/pages/2019-04-22---bulding-a-slack-bot-to-help-handle-large-numbers-of-prs/index.md

```diff
@@ -0,0 +1,127 @@
+
+>As obvious as this may seem, this economic view of product development is surprisingly uncommon. Instead, product developers create a large number of proxy objectives: increase innovation, improve quality, conform to plan, shorten development cycles, eliminate waste, etc. Unfortunately, they rarely understand how these proxy objectives quantitatively influence profits. Without this quantitative mapping, they cannot evaluate decisions that affect multiple interacting objectives.
+
+He dislikes proxies :-)
+
+We should measure what we actually care about — does some work contribute to the company’s mission and bottom line (directly or [indirectly](<https://www.amazon.com/dp/B004H4XD40/ref=dp-kindle-redirect?_encoding=UTF8&btkr=1>)).
+
+Clearly for Gatsby, we think PRs and the people writing the PRs are valuable — in the sense that our mission is to directly help build people and communities and to create tools that empower people and organizations. We want contributors because participating in Open Source directly helps their career and their contributions directly improves Gatsby which benefits everyone using it and the better Gatsby becomes, the more viable our commerical products & services become — which viability is what sustains and drives the whole process.
+
+## Managing Queues
+
+The next chapter is on “Managing Queues”. Really fun stuff.
+
+>In the last chapter, I showed you how to use an economic framework to guide your product development decisions. When we use such a framework, we often discover that cycle time has high economic leverage. If we can find cheap ways to reduce cycle time, our profits go up. But what causes long product development cycle time?
+
+He dives into the theory of queues (largely invented by a mathematician named Agner Karup Erlang (where have I heard that name before…🤔) at the Copehhangen Telephone Company who in 1909 was trying to determine how many phone lines were needed for a specific number of people) and how to manage queues — particularly when they get too long.
+
+The main principle from there we’ve taken to heart at Gatsby is principle 4: “**The Principle of High-Queue States: Most of the damage done by a queue is caused by high-queue states.**"
+
+“High-queue states” means in our context — too many PRs. PR count naturally flucuates as PRs come in semi-randomly and core maintainers have time to review and merge them. Sometimes these natural fluctuations mean the PR count soars to unusually high numbers. If not taken care of rapidly, this means the whole process will get bogged down.
+
+The rule we’ve adopted is that 120 PRs is a warning state when more people should shift over to working on PRs and 130 is red state when everyone should stop what they’re doing.
```

Comment:
```suggestion
The rule we’ve adopted is that 115 PRs is a warning state when more people should shift over to working on PRs and 130 is red state when everyone should stop what they’re doing to focus on PRs.
```

Also maybe mention that this is a _baseline_ that we intend to ensure this remains an accurate representation of work (e.g. we may tweak baseline to be 115 in the future).

Reply:
Yeah 👍 I changed it to say "current rule" to suggest it might change. It sounded a bit complex to explain why the rule might change in the future so didn't add any more explanation.